spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,256 @@
+
+		super.onInit();
+
+		Assert.notNull(this.debeziumEngineBuilder, "Failed to resolve Debezium Engine Builder. " +
+				"Debezium Engine Builder must either be set explicitly via constructor argument.");
+		Assert.notNull(this.executorService, "Invalid Executor Service. ");
+		Assert.notNull(this.headerMapper, "Header mapper can not be null!");
+
+		this.debeziumEngine = this.debeziumEngineBuilder
+				.notifying(new ChangeEventConsumer<byte[]>(this.headerMapper, this.contentType))
+				.build();
+	}
+
+	@Override
+	protected void doStart() {
+		if (this.latch.getCount() > 0) {
+			return;
+		}
+		this.latch = new CountDownLatch(1);
+		this.future = this.executorService.submit(() -> {
+			try {
+				this.debeziumEngine.run();
+			}
+			finally {
+				this.latch.countDown();
+			}
+		});
+		this.latch = new CountDownLatch(1);
+	}
+
+	@Override
+	protected void doStop() {
+		if (this.future.isDone()) {
+			return;
+		}
+		this.future.cancel(true);
+		try {
+			if (!this.latch.await(5, TimeUnit.SECONDS)) {
+				throw new IllegalStateException("Failed to stop " + this);
+			}
+		}
+		catch (InterruptedException ignored) {
+		}
+	}
+
+	@Override
+	public void destroy() {
+		super.destroy();
+		if (!this.executorServiceExplicitlySet) {
+			this.executorService.shutdown();
+		}
+		if (this.debeziumEngine != null) {
+			try {
+				this.debeziumEngine.close();
+			}
+			catch (IOException e) {
+				throw new UncheckedIOException("Debezium failed to close!", e);
+			}
+		}
+	}
+
+	void nextMessage(Message<?> message) {
+		this.sendMessage(message);
+	}
+
+	private final class ChangeEventConsumer<T> implements Consumer<ChangeEvent<T, T>> {
+
+		/**
+		 * Outbound message content type. Should be aligned with the {@link SerializationFormat} configured for the
+		 * {@link DebeziumEngine}.
+		 */
+		private final String contentType;
+
+		/**
+		 * Specifies how to convert Debezium change event headers into Message headers.
+		 */
+		private HeaderMapper<List<Header<T>>> headerMapper;
+
+		ChangeEventConsumer(HeaderMapper<List<Header<T>>> headerMapper, String contentType) {
+			this.headerMapper = headerMapper;
+			this.contentType = contentType;
+		}
+
+		@Override
+		public void accept(ChangeEvent<T, T> changeEvent) {
+			if (DebeziumMessageProducer.logger.isDebugEnabled()) {
+				DebeziumMessageProducer.logger.debug("[Debezium Event]: " + changeEvent.key());
+			}
+
+			Object key = changeEvent.key();
+			Object payload = changeEvent.value();
+			String destination = changeEvent.destination();
+
+			// When the tombstone event is enabled, Debezium serializes the payload to null (e.g. empty payload)
+			// while the metadata information is carried through the headers (debezium_key).
+			// Note: Event for none flattened responses, when the debezium.properties.tombstones.on.delete=true
+			// (default), tombstones are generate by Debezium and handled by the code below.
+			if (payload == null) {
+				payload = DebeziumMessageProducer.this.kafkaNull;
```

Comment:
I think we need to get rid of Kafka mentioning in this module.
Make this type of payload as `Optional.empty()` instead.